code/game/gamemodes/events/hivemind_invasion.dm

```diff
@@ -0,0 +1,35 @@
+//Hivemind is rogue AI that uses unknown nanotech to follow some strange objective
+//In fact, it's just hostile structures, wireweeds spreading event with some mobs
+//Requires hard teamwork at late stages, but easily can be handled at the beginning
+
+//All code stored in modules/hivemind
+
+
+/datum/storyevent/hivemind
+	id = "hivemind"
+	name = "Hivemind invasion"
+
+
+	event_type = /datum/event/hivemind
+	event_pools = list(EVENT_LEVEL_MAJOR = POOL_THRESHOLD_MAJOR*0.80)
+	tags = list(TAG_COMMUNAL, TAG_DESTRUCTIVE, TAG_NEGATIVE, TAG_SCARY)
+//============================================
+
+/datum/event/hivemind
+	announceWhen	= 32
+
+
+/datum/event/hivemind/announce()
+	level_seven_announcement()
+
+
+/datum/event/hivemind/start()
+	var/area/A = random_ship_area(filter_players = TRUE, filter_maintenance = TRUE, filter_critical = TRUE)
+	var/turf/T = A.random_space()
+	if(!T)
```

Comment:
This is giving up too easily. Have a look at a few other events for examples

Failing to find a turf is not uncommon, and the event should not randomly fail like that.
Setup a loop to make 100 attempts or so